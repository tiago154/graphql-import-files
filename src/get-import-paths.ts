/**
 * Checks for external files imported by the #import annotation in a file and returns the paths if any
 * @param {string} fileContent
 * @returns {string[] | []} The list of file paths with #import notation
 */
export const getImportPaths = (fileContent: string): string[] | [] =>
  fileContent
    .split(/\r?\n/)
    .filter((line) => line.match(/^#import\s/g))
    .map(line => line.match(/'.*'|".*"/g)!.toString().replace(/'|"/g, ''))
