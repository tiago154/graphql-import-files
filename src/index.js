const loadFile = require('./loadFile')
const loadFiles = require('./loadFiles')

module.exports = { loadFile, loadFiles }
