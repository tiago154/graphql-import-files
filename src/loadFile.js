const fs = require('fs')
const path = require('path')

/**
 * Place the path of the file to be imported, considering the root of the project.
 * @example ./schemas/file.graphql
 * @param {string} pathFile
 * @return {string} String of a scheme
 */
module.exports = pathFile =>
  fs.readFileSync(path.join(process.cwd(), pathFile), 'utf8')