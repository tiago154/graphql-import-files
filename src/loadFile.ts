import { readFileSync } from 'fs'
import { join } from 'path'

/**
 * Place the path of the file to be imported, considering the root of the project.
 * @example ./schemas/file.graphql
 * @param {string} pathFile
 * @return {string} String of a scheme
 */
export const loadFile = (pathFile: string) : string =>
  readFileSync(join(process.cwd(), pathFile), 'utf8')
