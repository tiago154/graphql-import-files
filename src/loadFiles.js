const fs = require('fs')
const path = require('path')
const glob = require('glob')
const { mergeTypeDefs } = require('@graphql-toolkit/schema-merging')


/**
 * Loads multiple graphql files, use .graphql or .gql.
 * @param {string} matchFiles - The glob patterns uses to detect graphql files
 * @return {DocumentNode} DocumentNode
 */
module.exports = matchFiles => {
  const files = glob.sync(matchFiles)

  const schema = files.map(file =>
    fs.readFileSync(path.join(process.cwd(), file), 'utf8')
  )

  return mergeTypeDefs(schema)
}