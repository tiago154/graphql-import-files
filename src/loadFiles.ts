import { readFileSync } from 'fs'
import { join } from 'path'

import glob from 'fast-glob'
import { mergeTypeDefs } from '@graphql-tools/merge'
import type { DocumentNode } from 'graphql'

/**
 * Loads multiple graphql files, use .graphql or .gql.
 * @param {string} matchFiles - The glob patterns uses to detect graphql files
 * @return {object} DocumentNode
 */
export const loadFiles = (matchFiles: string): DocumentNode => {
  const files = glob.sync(matchFiles)

  const schema = files.map(file =>
    readFileSync(join(process.cwd(), file), 'utf8')
  )

  return mergeTypeDefs(schema)
}
