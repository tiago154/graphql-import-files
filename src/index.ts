import { loadFile } from './loadFile'
import { loadFiles } from './loadFiles'

export { loadFile, loadFiles }
