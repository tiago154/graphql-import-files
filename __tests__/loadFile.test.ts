import { loadFile } from '../src'

describe('loadFile', () => {
  describe('when the file path is valid', () => {
    it('should return a valid schema', () => {
      const schema = loadFile('./__tests__/mocks/file/user.test.graphql')

      expect(schema).toMatchSnapshot()
    })
  })

  describe('when file path is valid and has file import annotation', () => {
    it('should valid schema', () => {
      const schema = loadFile('./__tests__/mocks/with-import/root.test.graphql')

      expect(schema).toMatchSnapshot()
    })
  })

  describe('when file path does not exist', () => {
    it('should an error when not finding the file', () => {
      expect(() => loadFile('./__tests__/mocks/file/error.graphql')).toThrow()
    })
  })
})
