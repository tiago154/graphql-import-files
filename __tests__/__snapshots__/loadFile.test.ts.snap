// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadFile success should valid schema 1`] = `
"type UserTest {
  id: Int!
  name: String!
  age: Int
  tags: [String!]!
}"
`;
